src/context/WalletProvider/MetaMask/components/Connect.tsx

```diff
@@ -58,9 +55,7 @@ export const MetaMaskConnect = ({ history }: MetaMaskSetupProps) => {
     }
 
     if (state.adapters && state.adapters?.has(KeyManager.MetaMask)) {
-      const wallet = (await state.adapters
-        .get(KeyManager.MetaMask)
-        ?.pairDevice()) as MetaMaskHDWallet
+      const wallet = await state.adapters.get(KeyManager.MetaMask)?.pairDevice()
```

Comment:
what happens if a user is on another, unsupported EVM chain, e.g. gnosis?

Reply:
this should probably be something like (pseudocode) `KnownChainIds.filter(chainId => fromChainId(chainId).chainNamespace === CHAIN_NAMESPACE.Ethereum)`?

Reply:
and similarly for all the other wallets - can prob be abstracted too?

Reply:
https://github.com/shapeshift/lib/pull/883 is the guy